app/models/openscap_result.rb

```diff
@@ -0,0 +1,76 @@
+class OpenscapResult < ApplicationRecord
+  belongs_to :container_image
+  has_one    :binary_blob,           :dependent => :destroy, :autosave => true, :as => :resource
+  has_many   :openscap_rule_results, :dependent => :destroy, :autosave => true
+
+  def self.openscap_available?
+    # needed only for travis
+    require 'openscap'
+    require 'openscap/ds/arf'
+    require 'openscap/xccdf/benchmark'
+    true
+  rescue LoadError
+    false
+  end
+
+  def raw=(value)
+    raise "Cannot invoke raw more than once." unless binary_blob.nil?
+    self.binary_blob = BinaryBlob.new(:name => 'openscap_compliance_arf', :data_type => 'XML')
+    binary_blob.binary = value
+    with_openscap_objects(binary_blob.binary) do |rule_results, benchmark_items|
+      create_results(rule_results, benchmark_items)
+    end
+  end
+
+  def html # Must be called after setting raw or it will explode
+    with_openscap_arf(binary_blob.binary) do |arf|
+      ascii8bit_to_utf8(arf.html)
+    end
+  end
+
+  private
+
+  def create_results(rule_results, benchmark_items)
+    rule_results.each do |openscap_id, result|
+      openscap_rule_results << OpenscapRuleResult.new(
+        :name            => ascii8bit_to_utf8(openscap_id),
+        :result          => ascii8bit_to_utf8(result.result),
+        :openscap_result => self,
+        :severity        => ascii8bit_to_utf8(benchmark_items[openscap_id].severity)
+      )
+    end
+  end
+
+  def ascii8bit_to_utf8(string)
+    string.to_s.encode('utf-8', :invalid => :replace, :undef => :replace, :replace => '_')
+  end
+
+  def with_openscap_arf(raw)
+    raise "no block given" unless block_given?
```

Comment:
not needed, ruby will complain and raise `LocalJumpError` if you call the method without a block